ghcide/src/Development/IDE/Types/Exports.hs

```diff
@@ -91,25 +99,32 @@ mkIdentInfos mod (AvailTC _ nn flds)
       ]
 
 createExportsMap :: [ModIface] -> ExportsMap
-createExportsMap = ExportsMap . Map.fromListWith (<>) . concatMap doOne
+createExportsMap modIface = do
+  let exportList = concatMap doOne modIface
+  ExportsMap (Map.fromListWith (<>) exportList) (buildModuleExportMap exportList)
   where
-    doOne mi = concatMap (fmap (second Set.fromList) . unpackAvail mn) (mi_exports mi)
-      where
-        mn = moduleName $ mi_module mi
+    doOne modIFace = do
+      let getModDetails = unpackAvail $ moduleName $ mi_module modIFace
+      concatMap (fmap (second Set.fromList) . getModDetails) (mi_exports modIFace)
```

Comment:
The efficient thing to do here is to change `unpackAvail` and `doOne` to return a triple `(moduleName, identifierText, IdentInfo)`

Reply:
addressed